### PR TITLE
Better docs for downward API

### LINKED ISSUE #10130: Add the "when" and "why" to use the Downward API
We need to add more content around the [Downward API](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/downward_api.md) to better clarify when and why to use it. We should also try to revise the "how" info we provide today to improve/clarify what the specific steps. 
(for example: Prerequisites are xyx. To use the downward API: 1. do this. 2....3...etc..).

Some details we should make clear:
 - When do i want containers to consume info about the system without coupling to k8s client or REST API?
 - Do we say "use the downward API when your containers need access to information about the cluster in which it resides"?
 - We should also tie together/link/mention the other content we have on using [environment variables](https://github.com/GoogleCloudPlatform/kubernetes/tree/master/examples/environment-guide) and the [downward api example](https://github.com/GoogleCloudPlatform/kubernetes/tree/master/examples/downward-api).

### PR DESCRIPTION
**Pull request by thockin**
Fixes #10130

@RichieEscarez @bgrant0607 @pmorie

### CODE REVIEW

**Comment by thockin** [comment]
ping @pmorie for a read-over

File: docs/downward_api.md [hunk 1]
```diff
@@ -1,28 +1,51 @@
 # Downward API
 
-The downward API allows containers to consume information about the system without coupling to the
+It is sometimes useful for a container to have information about itself, but we
+want to be careful not to over-couple containers to Kubernetes. The downward
+API allows containers to consume information about themselves or the system and
```
Comment (pmorie) [inline docs/downward_api.md]: @thockin What pieces of info that you can get to via the downward API qualify as about the system?

**Comment by pmorie** [comment]
@thockin We must have had some spooky action at a distance happening, since I just wrote about the same doc for OpenShift.

One suggestion, we should spell out which expressions give you which fields explicitly, otherwise LGTM.

**Comment by thockin** [comment]
I'm swamped - do you have time to fix this up if I merge it?

**Comment by pmorie** [comment]
@thockin yep, I have bandwidth

**Comment by pmorie** [comment]
@thockin did you mean to remove the label?

**Comment by nikhiljindal** [comment]
This one seems like it is ready to be merged.
Adding the ok-to-merge label

**Pull request merged**